inference.py

```diff
@@ -323,61 +284,77 @@ def main(params: dict):
-
-                        label_classes = np.unique(label)
-                        assert len(colors) >= len(label_classes), f'Not enough colors and class names for number of classes in output'
-                        # FIXME: color mapping scheme is hardcoded for now because of memory constraint; To be fixed.
-                        label_rgb = ind2rgb(label, colors)
-                        pred_rgb = ind2rgb(pred, colors)
-                        Image.fromarray((label_rgb).astype(np.uint8), mode='RGB').save(os.path.join(working_folder, 'label_rgb_' + inference_image.stem + '.png'))
-                        Image.fromarray((pred_rgb).astype(np.uint8), mode='RGB').save(os.path.join(working_folder, 'pred_rgb_' + inference_image.stem + '.png'))
-                        del label_rgb, pred_rgb
-                    pred = pred[np.newaxis, :, :]
+                    pred = pred[np.newaxis, :, :].astype(np.uint8)
                     inf_meta.update({"driver": "GTiff",
                                      "height": pred.shape[1],
                                      "width": pred.shape[2],
                                      "count": pred.shape[0],
                                      "dtype": 'uint8'})
-
                     with rasterio.open(inference_image, 'w+', **inf_meta) as dest:
                         dest.write(pred)
-        log_artifact(working_folder)
+        if len(gdf_) >= 1:
+            assert len(gdf_) == len(gpkg_name_), 'benchmarking unable to complete'
+            all_gdf = pd.concat(gdf_)
+            all_gdf.reset_index(drop=True, inplace=True)
+            gdf_x = gpd.GeoDataFrame(all_gdf)
+            gdf_x.to_file(working_folder.joinpath("benchmark.gpkg"), driver="GPKG", index=False)
+        # log_artifact(working_folder)
     time_elapsed = time.time() - since
     print('Inference and Benchmarking completed in {:.0f}m {:.0f}s'.format(time_elapsed // 60, time_elapsed % 60))
 
+
 if __name__ == '__main__':
     print('\n\nStart:\n\n')
-    parser = argparse.ArgumentParser(description='Inference and Benchmark on images using trained model')
-    parser.add_argument('param_file', metavar='file',
+    parser = argparse.ArgumentParser(usage="%(prog)s [-h] [YAML] [-i MODEL IMAGE] ",
+                                     description='Inference and Benchmark on images using trained model')
+
+    parser.add_argument('-p', '--param', metavar='yaml_file', nargs=1,
                         help='Path to parameters stored in yaml')
+    parser.add_argument('-i', '--input', metavar='model_pth img_dir', nargs=2,
+                        help='model_path and image_dir')
     args = parser.parse_args()
-    params = read_parameters(args.param_file)
+
+    if args.param:
+        params = read_parameters(args.param_file)
+    elif args.input:
+        model = Path(args.input[0])
+        image = args.input[1]
+
+        checkpoint = load_checkpoint(model)
+        params = checkpoint['params']
```

Comment:
maybe raise an error with custom message if key "params" not in model checkpoint. e.g.: "No parameters found in checkpoint. Use GDL version 1.3 or more."

Reply:
okay, more explicit than the inbuilt error message